packages/helix-shared-tokencache/src/FSCachePlugin.js

```diff
@@ -38,34 +42,75 @@ export class FSCachePlugin {
     }
   }
 
-  /**
-   * @param {TokenCacheContext} cacheContext
-   * @returns {Promise<boolean>} if cache was updated
-   */
-  async beforeCacheAccess(cacheContext) {
+  async #loadData() {
     const { log, filePath } = this;
     try {
-      cacheContext.tokenCache.deserialize(await fs.readFile(filePath, 'utf-8'));
-      return true;
+      let raw = await fs.readFile(filePath, 'utf-8');
+      const data = JSON.parse(raw);
+      if (data.cachePluginMetadata) {
+        this.meta = data.cachePluginMetadata;
+        delete data.cachePluginMetadata;
+        raw = JSON.stringify(data);
+      } else {
+        this.meta = {};
+      }
+      this.data = data;
+      return raw;
     } catch (e) {
       if (e.code !== 'ENOENT') {
         // only log warnings if file exists, otherwise ignore
         log.warn('FSCachePlugin: unable to deserialize', e);
       }
     }
+    this.data = null;
+    return null;
+  }
+
+  async #saveData() {
+    const { filePath } = this;
+    const data = this.data || {};
+    if (Object.keys(this.meta || {}).length) {
+      data.cachePluginMetadata = this.meta;
```

Comment:
add `cachePluginMetadata` before save

Reply:
possibly a stupid question, but should you also remove `data.cachePluginMetadata` after saving?

Reply:
good point